EKF/ekf_helper.cpp

```diff
@@ -99,35 +90,52 @@ bool Ekf::resetVelocity()
 		if(_params.fusion_mode & MASK_ROTATE_EV){
 			_ev_vel = _R_ev_to_ekf *_ev_sample_delayed.vel;
 		}
-		_state.vel = _ev_vel;
+		resetVelocityTo(_ev_vel);
 		P.uncorrelateCovarianceSetVariance<3>(4, _ev_sample_delayed.velVar);
+
 	} else {
 		ECL_INFO_TIMESTAMPED("reset velocity to zero");
 		// Used when falling back to non-aiding mode of operation
-		_state.vel(0) = 0.0f;
-		_state.vel(1) = 0.0f;
+		resetHorizontalVelocityTo(Vector2f{0.f, 0.f});
 		P.uncorrelateCovarianceSetVariance<2>(4, 25.0f);
 	}
 
-	// calculate the change in velocity and apply to the output predictor state history
-	const Vector3f velocity_change = _state.vel - vel_before_reset;
+	return true;
+}
+
+void Ekf::resetVelocityTo(const Vector3f &new_vel) {
+	resetHorizontalVelocityTo(Vector2f(new_vel.xy()));
+	resetVerticalVelocityTo(new_vel(2));
+}
+
+void Ekf::resetHorizontalVelocityTo(const Vector2f &new_horz_vel) {
+	const Vector2f delta_horz_vel = new_horz_vel - Vector2f(_state.vel.xy());
+	_state.vel(0) = new_horz_vel(0);
```

Comment:
Can't we use `_state.vel.xy() = new_horz_vel` here?

Reply:
I am not sure. I will try after adding the tests.

Reply:
This is working. Will change it